### PR TITLE
More shards

### PR DESCRIPTION
**Pull request by dnfield**
- Cache the build output for tool tests
- Shard out integration tests.

### CODE REVIEW

**Comment by dnfield** [comment]
My have prematurely marked this ready for review-  looks like Windows isn't happy with something.  Linux and Mac should be good though.

File: packages/flutter_tools/lib/src/base/build.dart [hunk 1]
```diff
@@ -248,7 +248,7 @@ class AOTSnapshotter {
     final List<String> commonBuildOptions = <String>['-arch', targetArch, '-miphoneos-version-min=8.0'];
 
     final String assemblyO = fs.path.join(outputPath, 'snapshot_assembly.o');
-    final RunResult compileResult = await xcode.cc(commonBuildOptions.toList()..addAll(<String>['-c', assemblyPath, '-o', assemblyO]));
+    final RunResult compileResult = await xcode.cc(commonBuildOptions.toList()..addAll(<String>['-c', assemblyPath, '-o', assemblyO, '-fembed-bitcode']));
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/base/build.dart]: This does not belong in this PR, will remove.

File: packages/flutter_tools/lib/src/base/build.dart [hunk 2]
```diff
@@ -262,6 +262,7 @@ class AOTSnapshotter {
         '-Xlinker', '-rpath', '-Xlinker', '@executable_path/Frameworks',
         '-Xlinker', '-rpath', '-Xlinker', '@loader_path/Frameworks',
         '-install_name', '@rpath/App.framework/App',
+        '-fembed-bitcode',
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/base/build.dart]: Ditto

File: .cirrus.yml [hunk 3]
```diff
@@ -185,6 +204,52 @@ task:
...
+    - bin\flutter.bat update-packages
+    - git fetch origin master
+  build_runner_cache:
+    fingerprint_script: type bin\cache\dart-sdk\version
+    folder: .\packages\flutter_tools\.dart_tool
+    populate_script: powershell dev\bots\flutter_tool_build_cache.ps1
+  test_all_script:
+    - bin\cache\dart-sdk\bin\dart.exe --enable-asserts dev\bots\test.dart
+  matrix:
+    - name: tool_tests-windows
```
Comment (digiter) [inline .cirrus.yml]: What's the different between "tool_tests" and "tool_tests_integration"? Consider adding some comment to clarify.

File: .cirrus.yml [hunk 4]
```diff
@@ -285,6 +346,49 @@ task:
...
 
+task:
+  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
+  osx_instance:
+    image: mojave-xcode-10.1
+  depends_on:
+    - analyze
+  env:
+    CIRRUS_WORKING_DIR: "/tmp/flutter sdk"
+    COCOAPODS_DISABLE_STATS: true
```
Comment (digiter) [inline .cirrus.yml]: Maybe it's more readable to name it "COCOAPODS_DISABLED"?

Comment (dnfield) [reply .cirrus.yml]: That's an environment variable specifically for Cocoapods, we can't change the name of it.  It disables metrics reporting, which is slow and not needed for CI anyway.

File: .cirrus.yml (hunk 3 above)
Comment (dnfield) [reply .cirrus.yml]: tool_tests runs everything in the `packages/flutter_tools/test` except the integration folder.

tool_tests_integration runs everything in the `packages/flutter_tools/test/integration` folder.  I can add some comments, still trying to see what caching strategy is helping or hurting.

**Comment by dnfield** [comment]
This doesn't really seem to help enough.  Going to try one more strategy here.

**Comment by digiter** [comment]
Please ping when this PR becomes ready.

**Comment by dnfield** [comment]
This is ready for review - the tool test shards are now all below ~25 minutes.

File: packages/flutter_tools/test/commands/create_test.dart [hunk 5]
```diff
@@ -57,1064 +57,1067 @@ void main() {
     tryToDelete(tempDir);
   });
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands/create_test.dart]: Here I would place the tag on the file instead of adding another layer of nesting

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by digiter — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
flutter-build lies, the build is green.  merging.
